core/src/main/java/io/fabric8/maven/core/handler/ContainerHandler.java

```diff
@@ -83,11 +84,28 @@ public ContainerHandler(Properties configurationProperties, Artifact artifact, E
         return ret;
     }
 
+    private List<EnvVar> getEnvVars(ResourceConfig config) {
+        List<EnvVar> envVars = KubernetesResourceUtil.convertToEnvVarList(config.getEnv().orElse(Collections.emptyMap()));
+
+        // TODO: This should go into an extra enricher so that this behaviour can be switched on / off
+        envVars.add(0,
+            new EnvVarBuilder()
+                .withName("KUBERNETES_NAMESPACE")
+                .withNewValueFrom()
+                  .withNewFieldRef()
+                     .withFieldPath("metadata.namespace")
+                  .endFieldRef()
+                .endValueFrom()
+                .build());
+
+        return envVars;
+    }
+
 
     private String getImagePullPolicy(ResourceConfig config) {
         String pullPolicy = config.getImagePullPolicy();
         if (StringUtils.isBlank(pullPolicy) &&
-            this.artifact.getVersion() != null && this.artifact.getVersion().endsWith("SNAPSHOT")) {
+            this.groupArtifactVersion.getVersion() != null && this.groupArtifactVersion.getVersion().endsWith("SNAPSHOT")) {
```

Comment:
How about adding a method in `GroupArtifactVersion` that is called `isSnapshot()`

Reply:
Good point, let me introduce that.